test/test_torch.py

```diff
@@ -13451,8 +13326,126 @@ def add_neg_dim_tests():
         assert not hasattr(_TestTorchMixin, test_name), "Duplicated test name: " + test_name
         setattr(_TestTorchMixin, test_name, make_neg_dim_test(name, tensor_arg, arg_constr, types, extra_dim))
 
-add_neg_dim_tests()
+# Device-agnostic tests. Instantiated below and not run directly.
```

Comment:
err, agnostic isn't the word I would use for these tests.  Device-generic?  Generic-Device-aware?

Reply:
Device-generic sounds good.